### ISSUE TITLE
Migrations failing in 0.50.11 with: constraint "idx_uniq_field_table_id_parent_id_name" of relation "metabase_field" does not exist

### ISSUE THREAD

**Issue by michael-gratton** [metabase/metabase#45334]
### Describe the bug

Upgraded to 0.50.11 self-hosted, started getting the logs below. Downgrading to 0.50.10 fixed it.

```
2024-07-10 08:46:44,933 ERROR metabase.core :: Metabase Initialization FAILED
liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v49.2024-06-27T00:00:05::calherries:
     Reason: liquibase.exception.DatabaseException: ERROR: constraint "idx_uniq_field_table_id_parent_id_name" of relation "metabase_field" does not exist [Failed SQL: (0) ALTER TABLE "public"."metabase_field" DROP CONSTRAINT "idx_uniq_field_table_id_parent_id_name"]
```

### To Reproduce

1. Run 0.50.10
2. Deploy 0.50.11

### Expected behavior

The migration should not fail. Metabase should start.

### Logs

See above

### Information about your Metabase installation

```
0.50.11, self hosted as AWS ECS Fargate container using RDS Aurora PostreSQL for application database.
```

### Severity

CRITICAL

**Comment by michael-gratton** [comment]
Likely caused by https://github.com/metabase/metabase/pull/45246 ?

**Comment by calherries** [comment]
@michael-gratton thanks for raising this so quickly. That's very strange, that migration has preconditions to check the existence of the constraint before trying to drop it. Can you try upgrading again to 0.50.11 and tell us if it succeeds? Downgrading adds the constraint again, so you shouldn't encounter issues a second time.

**Comment by michael-gratton** [comment]
@calherries Can't roll forward again to 0.50.11 at the moment, since it is in production. However re-deploying to 0.50.10 didn't add `idx_uniq_field_table_id_parent_id_name` back to `metabase_field`, so I don't see rolling forward again helping.

**Comment by qnkhuat** [comment]
@michael-gratton a fix will be included in the next release; in the meantime, you can work around it by manually adding the index and then migrating again.

```sql
CREATE UNIQUE INDEX idx_uniq_field_table_id_parent_id_name ON "metabase_field" ("table_id", "name", "parent_id")
```

**Comment by calherries** [comment]
@michael-gratton can you share the results of this query? I'm trying to understand why there was no `idx_uniq_field_table_id_parent_id_name` constraint in your app DB.

```
SELECT 'CONSTRAINT' as type,
       conname as name,
       pg_get_constraintdef(c.oid) as definition
FROM pg_constraint c
JOIN pg_namespace n ON n.oid = c.connamespace
JOIN pg_class cl ON cl.oid = c.conrelid
WHERE cl.relname = 'metabase_table'

UNION ALL

SELECT 'INDEX' as type,
       i.relname as name,
       pg_get_indexdef(i.oid) as definition
FROM pg_index x
JOIN pg_class c ON c.oid = x.indrelid
JOIN pg_class i ON i.oid = x.indexrelid
JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE c.relname = 'metabase_table'

ORDER BY type, name;
```

**Comment by calherries** [comment]
Thanks! That lead me to the likely culprit. The migration that adds the `idx_uniq_field_table_id_parent_id_name` included a precondition that would not add the constraint in the case where this query results in 0 rows:
```
SELECT count(*) FROM METABASE_FIELD GROUP BY TABLE_ID, PARENT_ID, NAME HAVING count(*) > 1
```
https://github.com/metabase/metabase/blob/cf92c8bff67bd508741c05b90db4218b827d53a2/resources/migrations/000_legacy_migrations.yaml#L5663-L5692

Your app DB failed to pass this precondition and the constraint was never added.

The same is true of `metabase_table`: there's a precondition on the constraints that add `idx_uniq_table_db_id_schema_name` and `idx_uniq_table_db_id_schema_name_2col`:

https://github.com/metabase/metabase/blob/cf92c8bff67bd508741c05b90db4218b827d53a2/resources/migrations/000_legacy_migrations.yaml#L5632-L5662

 You app DB also doesn't have these constraints.

The lack of `metabase_field` constraints will be addressed by [v0.50.12](https://github.com/metabase/metabase/milestone/252) but the lack of `metabase_table` constraints should be addressed. Not having those constraints increases the likelihood of bugs in the future.
